core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/scheduling/resourcePolicies/ResourceAllocation.scala

```diff
@@ -0,0 +1,40 @@
+package edu.uci.ics.amber.engine.architecture.scheduling.resourcePolicies
+
+import edu.uci.ics.amber.engine.architecture.scheduling.Region
+import edu.uci.ics.amber.engine.common.AmberConfig
+import edu.uci.ics.texera.workflow.common.workflow.PhysicalPlan
+
+import scala.collection.mutable.ListBuffer
+
+object ResourceAllocation {
+  def apply(region: Region): ResourceAllocation = {
+    ResourceAllocation(region)
+  }
+}
+
+case class ResourceAllocation(region: Region) {
+
+  private def physicalOpIds = region.physicalOpIds.filter(_.logicalOpId.id.contains("PythonUDF"))
+
+  def allocation(
```

Comment:
1. please use verb such as `allocate` instead of noun for function names.
2. the function should take `region` as its parameter (and other necessary information).
3. the function should return a modified Region (with RegionConfig set) and a double for cost. 
4. add documents.